src/debugAdapter/goDebug.ts

```diff
@@ -692,6 +761,25 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	protected launchRequest(response: DebugProtocol.LaunchResponse, args: LaunchRequestArguments): void {
+		args.kind = 'launch';
+		if (!args.program) {
+			this.sendErrorResponse(response, 3000, 'Failed to continue: The program attribute is missing in the debug configuration in launch.json');
+			return;
+		}
+		this.initLaunchAttachRequest(response, args);
+	}
+
+	protected attachRequest(response: DebugProtocol.AttachResponse, args: AttachRequestArguments): void {
+		args.kind = 'attach';
+		if (args.mode === 'attach' && !args.processId) {
+			this.sendErrorResponse(response, 3000, 'Failed to continue: the processId attribute is missing in the debug configuration in launch.json');
+		} else if (args.mode === 'connect' && !args.port) {
```

Comment:
It seems like on line 686, we are assigning random ports to `args` if `args.port` is falsy. Do we need this check then?

Reply:
When attaching the user must specify the port to connect to.  The random port number is used in the `launch` scenario.